contracts/Core/VoteManager.sol

```diff
@@ -142,4 +157,13 @@ contract VoteManager is Initializable, ACL, VoteStorage {
         // epoch -> asset -> stakeWeight
         return (totalInfluenceRevealed[epoch][assetId]);
     }
+
+    function _getRandaoHash() internal view returns (bytes32) {
```

Comment:
Hey @hrishikeshio 
IMO this would not work for the reveal - isAssetAllotedToStaker

As per this code, we are having secrets of this epoch as seed, which is being updated in the same state where it is being used.
It should be the secret of the previous epoch right.
Epoch 1
secrets[4] = [A,B,C,D]

Epoch 2 
Staker 1 reveals with secret E and considering seed [A,B,C,D]
secrets[4] = [E,B,C,D]

Now for the next stakers, they won't be sure on present status of seed, as it's updated on each stakers reveal, and there is no defined order for them to reveal

Reply:
Maybe we can use stakers commitments as seed secret and commitment both have the same datatype bytes32, so theoretically it's possible.
Also this would solve another issue of knowing seed before commit as well

Reply:
Correct. But we are going to disable assign assets randomly right?

Reply:
If we use commitments as seed, all stakers would have to collude at once to manipulate prng

Reply:
no you can again wait till everyone has committed and then choose commitment which is favourable to you.

Reply:
> no you can again wait till everyone has committed and then choose commitment which is favourable to you.

Oh Yes